### PR TITLE
Initial creation of the SKY UX 4 version of Barkbaud.

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,13 +9,13 @@ Blackbaud SKY API / SKY UX sample application.
...
 - A dashboard provides a listing of dogs that are looking for a good home. Selecting a dog will take you to a page which lists the animal's owner and medical history. Biographies, owner and medical history for each animal are stored in a MongoDB database at [mongoDB](https://www.mongodb.com/).
-- Blackbaud SKY API provides access to the constituent data. The application then uses the API to search for a constituent and retrieve the constituent ID which is used to pair the constituent record to a dog within mLab.
-- Medical history is stored as a subdocument for each dog in mLab. Upon adding medical history, the user has the option of storing the information as a note on the current owner's Raiser's Edge NXT constituent record.
+- Blackbaud SKY API provides access to the constituent data. The application then uses the API to search for a constituent and retrieve the constituent ID which is used to pair the constituent record to a dog within mongoDb.
+- Medical history is stored as a subdocument for each dog in MongoDb. Upon adding medical history, the user has the option of storing the information as a note on the current owner's Raiser's Edge NXT constituent record.
```
Comment (michael-tims) [inline README.md]: `MongoDb` here, but previous sentence is `mongoDb`.  Should they both have uppercase `M`?

File: skyux-spa-ui/src/app/auth/auth.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  it('should do something', () => {
+    const fixture = TestBed.createComponent(AuthComponent);
+
+    fixture.detectChanges();
+
+    expect(true).toBe(false);
+  });
+
+});
```
Comment (michael-tims) [inline skyux-spa-ui/src/app/auth/auth.component.spec.ts]: Should we just delete this file?  and probably any other `spec.ts` files?

File: skyux-spa-ui/src/app/dogs/dogs-list.component.html [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      disableMargin="true"
+    >
+      <sky-row>
+        <sky-column screenXSmall="7">
+          <h3 class="app-dog-card-title">
+          <a routerLink="/dogs/{{ dog._id }}"
+                                    queryParamsHandling="preserve">
+                                      {{ dog.name }}
+                                    </a>
+                                  </h3>
```
Comment (michael-tims) [inline skyux-spa-ui/src/app/dogs/dogs-list.component.html]: Fix indentation

File: skyux-spa-ui/src/app/dogs/dogs-list.component.html [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  <sky-card-actions>
+    <a
+      [routerLink]="[dog._id]"
+      class="sky-btn sky-btn-default"
+    >
+      {{ 'btn_profile' | skyAppResources }}
+    </a>
+  </sky-card-actions>
+</sky-card>
+<!--
```
Comment (michael-tims) [inline skyux-spa-ui/src/app/dogs/dogs-list.component.html]: Remove commented code if not needed anymore.

**Review by michael-tims — COMMENTED** [review]
A few suggestions

File: package.json [hunk 5]
```diff
@@ -4,15 +4,15 @@
...
   "main": "index.js",
   "scripts": {
-    "build": "grunt build",
     "setup": "node ./bin/setup.js",
+    "dev": "nodemon ./bin/start.js",
     "start": "node ./bin/start.js",
     "lint": "node ./node_modules/.bin/eslint server",
     "test": "npm run lint",
-    "heroku-postbuild": "echo Skip build on Heroku"
+    "build": "cd skyux-spa-ui && npm i && npm i -g @skyux-sdk/cli@next && skyux build"
```
Comment (Blackbaud-SteveBrush) [inline package.json]:
```suggestion
    "build": "cd skyux-spa-ui && npm i && npx -p @skyux-sdk/cli skyux build"
```

Comment (Blackbaud-SteveBrush) [reply package.json]: ☝️ You can use `npx` to run a package globally (without installing it on the system). This is good practice for running global scripts on a CI.

File: package.json [hunk 6]
```diff
@@ -4,15 +4,15 @@
...
   },
   "dependencies": {
-    "async": "2.6.1",
+    "@skyux/layout": "4.3.1",
```
Comment (Blackbaud-SteveBrush) [inline package.json]: Shouldn't this be in the `skyux-spa-ui` `package.json` file?

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
When I ran `npm install` I encountered quite a few security vulnerabilities. Possible to fix those?

**Review by Blackbaud-ChristopherRodgers — DISMISSED** [review]

**Review by d4nleonard — APPROVED** [review]

**Review by Blackbaud-ChristopherRodgers — APPROVED** [review]

**Pull request merged**